chapter05/jxp/ch5ex1.pl

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/perl
+use v5.10;
+
+	my %total_bytes;
+	my $filename='coconet.dat';
+	open(my $fh, $filename);
+    while (my $row = <$fh>) {
+      my ($source, $destination, $bytes) = split '\t', $row;
+      $total_bytes{$source}{$destination} += $bytes;
+    }
+	
+	my %sum_total;
+	
+	for my $source ( keys %total_bytes) {
+      for my $destination ( keys %{ $total_bytes{$source} }) {
+         $sum_total{$source} += $total_bytes{$source}{$destination};
+      }
+    }
+	
+    for my $source (reverse sort { $sum_total{$a} <=> $sum_total{$b} } keys %sum_total) {
```

Comment:
Readability win! @alexdeas might have an opinion on the efficiency of it, though :smile: